fe/fe-core/src/main/java/org/apache/doris/planner/EsScanNode.java

```diff
@@ -224,7 +224,7 @@ private void assignBackends() throws UserException {
                 List<Backend> candidateBeList = Lists.newArrayList();
                 if (usingRandomBackend) {
                     for (int i = 0; i < numBe; ++i) {
-                        candidateBeList.add(backendList.get(beIndex++ % numBe));
+                        candidateBeList.add(backendList.get(beIndex++ % backendList.size()));
```

Comment:
```suggestion
                        candidateBeList.add(backendList.get(beIndex++ % size));
```
obtain size = backendList.size() in other place